content/blog/2024/12/how-to-migrate-from-idf-bootloader-to-mcuboot-using-esp-self-reflasher/index.md

```diff
@@ -0,0 +1,271 @@
+
+This tutorial demonstrates how ESP Self-Reflasher component can be used to migrate an IDF-bootloader-based application (OTA capable) to a MCUboot-based-bootloader application. The ESP Self-Reflasher component is used on an middle step application for enabling the device to "reflash" its own firmware.
+
+---
+**Note**
+The MCUboot compatible application used in this guide is a NuttX RTOS application, but could be other like a Zephyr RTOS application, for instance.
+
+---
+
+The overall process described in this guide can be illustrated as:
+
+```mermaid
+flowchart LR
+    A[IDF bootloader + IDF application with OTA capabilities] -->|OTA process| B[Reflashing application with ESP Self-Reflasher component]
+    B -->|Reflashing process| C(MCUboot bootloader + NuttX application)
+```
+
+## ESP Self Reflasher component
+
+The [ESP Self-Reflasher](https://components.espressif.com/components/espressif/esp-self-reflasher/versions/1.0.0) is an ESP-IDF component that allows the device to be "reflashed" with new firmware. This is achieved through one of the following ways of operation: downloading the reflashing image or embedding the reflashing image. The component has examples for each that will be used on this guide.
+
+## Environment pre requisites
+
+Besides ESP-IDF ([Get started guide here](https://docs.espressif.com/projects/esp-idf/en/stable/esp32/get-started/index.html)), the following are required:
+
+- ESP Self-Reflasher: clone it to `<IDF_DIR>/components` with Git:
+
+```bash
+cd <IDF_DIR>/components
+git clone https://github.com/espressif/esp-self-reflasher.git
+```
+
+- NuttX: in order to build the application that will be migrated to, set the NuttX workspace. [Get started guide here](https://developer.espressif.com/blog/nuttx-getting-started/).
+
+- MCUboot: in the case of building the bootloader standalone, follow the instructions [here](https://docs.mcuboot.com/readme-espressif.html).
+
+## Step-by-step Guide
+
+### Building the reflashing images (MCUboot and NuttX application)
+
+With the NuttX workspace ready, prepare the NSH (NuttShell) configuration with MCUboot compatibility, which is available under the `mcuboot_nsh` defconfig. This is the build that will be used as the *final target* of the reflashing process.
+
+```bash
+cd <NUTTX_DIR>
+./tools/configure.sh esp32-devkitc:mcuboot_nsh
+```
+
+---
+**Note**
+
+It is also possible to manually configure other NuttX applications for building it with MCUboot compatibility. Enter the menuconfig:
+
+```bash
+make menuconfig
+```
+
+Enter the menu `System Type -> Bootloader and Image Configuration` and enable the `Enable MCUboot-bootable format` option.
+
+---
```

Comment:
Why are you placing this information in a note? It seems to be as important and as relevant as any other information.

Also suggest optimizing the formatting:

````suggestion
You can also manually configure other NuttX applications to be built with MCUboot compatibility:

- In your NuttX application, enter the menuconfig:
  ```bash
  make menuconfig
  ```
- In the menu `System Type -> Bootloader and Image Configuration`, enable the option `Enable MCUboot-bootable format`
````